### PR TITLE
Stats: Update paywall upsell copy for modules

### PR DESCRIPTION
**Pull request by dognose24**
Related to https://github.com/Automattic/red-team/issues/95

## Proposed Changes

* Update existing copy of `StatsCardUpsellJetpack` for Stats modules.
* Add upsell copy for paywalled Stats modules and Stats cards on Traffic and Insights pages.

## Why are these changes being made?

* Apply copies according to the design for the restricted dashboard project: p1HpG7-tjV-p2.

## Testing Instructions

* Spin this change up with the Calypso Live branch.
* Prepare a Jetpack site **_without_** any Stats commercial or Jetpack plan purchase.
* Mark the site as **_commercial_** forcibly.
* Navigate to Stats > `Traffic` page with a feature flag: `?flags=stats/restricted-dashboard`.
* Ensure the paywalled modules display with copies and links according to the design.

截圖 2024-07-10 下午11 13 10
截圖 2024-07-10 下午11 13 24
截圖 2024-07-10 下午11 16 41

* Navigate to Stats > `Insights` page with a feature flag: `?flags=stats/restricted-dashboard`.
* Ensure the paywalled modules and cards display with copies and links according to the design.

截圖 2024-07-10 下午11 12 59
截圖 2024-07-10 下午11 12 50

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-card-upsell/stats-card-upsell-jetpack.tsx [hunk 1]
```diff
@@ -2,45 +2,104 @@ import { recordTracksEvent } from '@automattic/calypso-analytics';
...
+	);
+};
+
 const useUpsellCopy = ( statType: string ) => {
 	const translate = useTranslate();
 	switch ( statType ) {
 		case STATS_FEATURE_DATE_CONTROL:
 			return translate( 'Compare different time periods to analyze your site’s growth.' );
 		case STATS_FEATURE_UTM_STATS:
+			return translate( 'Generate UTM parameters and track your campaign performance data.' );
```
Comment (grzegorz-cp) [inline client/my-sites/stats/stats-card-upsell/stats-card-upsell-jetpack.tsx:62]: Can we pass those strings from specific components? We would avoid this growing switch, and all translations would be closer to the source that needs them.

Comment (dognose24) [reply client/my-sites/stats/stats-card-upsell/stats-card-upsell-jetpack.tsx:62]: > Can we pass those strings from specific components?

Do you mean passing from the module component that uses the upsell? I am not sure receiving the string from the respective component would benefit management unless we have already used the exact string.

> avoid this growing switch

We can use object keys for the `statType` to map the corresponding string to alter the growing switch cases.

> all translations would be closer to the source that needs them

I think consolidating the upsell copy inside the upsell component would be straightforward. Furthermore, in that case, we may also need to prepare the string for `StatsCardUpsellWPCOM` all in the source module, which would be noise.

**Review by kangzj — APPROVED** [review]
LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16216392

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @dognose24 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16216392).

**Pull request merged**